graphql-server/src/schemas/neo4j.ts

```diff
@@ -118,6 +94,135 @@ function buildResolvers(ogm: OGM, apolloClient: typeof ApolloClient) {
   };
 }
 
+async function updateSampleMetadata(
+  sampleManifest: any,
+  changesByPrimaryId: any,
+  ogm: OGM,
+  apolloClient: typeof ApolloClient
+) {
+  Object.keys(changesByPrimaryId).forEach((primaryId: string) => {
+    sampleManifest[primaryId] = changesByPrimaryId[primaryId];
+  });
+
+  // remove 'status' from sample metadata to ensure validator and label
+  // generator use latest status data added during validation process
+  delete sampleManifest["status"];
+
+  // add isCmoSample to sample's 'additionalProperties' if not already present
+  // this is to ensure that cmo samples get sent to the label generator after validation
+  // since some of the older SMILE samples do not have this additionalProperty set
+  if (sampleManifest["additionalProperties"]["isCmoSample"] == null) {
+    const requestId = sampleManifest["additionalProperties"]["igoRequestId"];
+    let req = ogm.model("Request");
+    const rd = await req.find({
+      where: { igoRequestId: requestId },
+    });
+    sampleManifest["additionalProperties"]["isCmoSample"] =
+      rd[0]["isCmoRequest"].toString();
+  }
+
+  // fire and forget
+  publishNatsMessage(
+    props.pub_validate_sample_update,
+    JSON.stringify(sampleManifest)
+  );
+
+  let sample = ogm.model("Sample");
+
+  await sample.update({
+    where: { smileSampleId: sampleManifest.smileSampleId },
+    update: { revisable: false },
+  });
+
+  const updatedSamples = await apolloClient.query({
+    query: SamplesDocument,
+    variables: {
+      where: {
+        smileSampleId: sampleManifest.smileSampleId,
+      },
+      hasMetadataSampleMetadataOptions2: {
+        sort: [{ importDate: SortDirection.Desc }],
+        limit: 1,
+      },
+    },
+  });
+
+  Object.keys(changesByPrimaryId).forEach((key: string) => {
+    updatedSamples.data.samples[0].hasMetadataSampleMetadata[0][key] =
+      changesByPrimaryId[key];
+  });
+
+  return updatedSamples;
+}
+
+async function updateSampleBilling(
```

Comment:
This function packages the data updates into the JSON schema expected by `smile-server` and then send the JSON as a message to NATS:

- primaryId - string
- billed - boolean (true/false)
- billedBy - string
- costCenter - string

This function will likely be modified in my next PR to correctly handle optimistic update.